src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const previousStateVers = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        copyState = Object.assign(copyState, action.extraData);
```

Comment:
You don't need to overwrite the object every time.

[The Object.assign() static method copies all enumerable own properties from one or more source objects to a target object. It returns the modified target object.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)